ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/Serialization/Base64UrlJsonConverter.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Rest.Serialization
+{
+    public class Base64UrlJsonConverter : JsonConverter
+    {
+        /// <summary>
+        /// Converts a byte array to a Base64Url encoded string
+        /// </summary>
+        /// <param name="input">The byte array to convert</param>
+        /// <returns>The Base64Url encoded form of the input</returns>
+        private static string ToBase64UrlString(byte[] input)
+        {
+            if (input == null)
+                throw new ArgumentNullException("input");
+
+            return Convert.ToBase64String(input).TrimEnd('=').Replace('+', '-').Replace('/', '_');
+        }
+
+        /// <summary>
+        /// Converts a Base64Url encoded string to a byte array
+        /// </summary>
+        /// <param name="input">The Base64Url encoded string</param>
+        /// <returns>The byte array represented by the enconded string</returns>
+        private static byte[] FromBase64UrlString(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+                throw new ArgumentNullException("input");
+
+            return Convert.FromBase64String(Pad(input.Replace('-', '+').Replace('_', '/')));
+        }
+
+        /// <summary>
+        /// Adds padding to the input
+        /// </summary>
+        /// <param name="input"> the input string </param>
+        /// <returns> the padded string </returns>
+        private static string Pad(string input)
+        {
+            var count = 3 - ((input.Length + 3) % 4);
+
+            if (count == 0)
+            {
+                return input;
+            }
+
+            return input + new string('=', count);
+        }
+
+        public override bool CanConvert(Type objectType)
+        {
+            if (objectType == typeof(byte[]))
```

Comment:
Yep, JsonConverter.DeserializeObject handles IList and dictionary cases correctly - it will just use the ReadJson function on the values of the inner type, not the collection as a whole. I've added two tests for both array and dictionary cases to make verify this is the case in my [latest commit](https://github.com/Azure/autorest/pull/878/commits/3e4df712343c7f3f0614d66133910efb65d319e0) . Thanks!